internal/service/project/resource_project_test.go

```diff
@@ -429,3 +858,106 @@ func TestAccProjectRSProject_withInvalidLimitNameOnUpdate(t *testing.T) {
 		},
 	})
 }
+
+func createDataFederationLimit(limitName string) admin.DataFederationLimit {
+	return admin.DataFederationLimit{
+		Name: limitName,
+	}
+}
+
+type MockProjectService struct {
+	MockResponses []ProjectResponse
```

Comment:
+1 , for instance testify ElementsMatch ignores order: https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L1039

Reply:
@lantoli @AgustinBettati Good idea, I changed to use `testify` library mocks, using this we can mock the API calls setting specific return values, instead of relying on the order of the calls to the API. Let me know what you think!